parser.go

```diff
@@ -651,6 +651,30 @@ func (p *parseState) addArgs(args ...string) error {
 	for len(p.positional) > 0 && len(args) > 0 {
 		arg := p.positional[0]
 
+		if !arg.isRemaining() && len(arg.Choices) != 0 {
+			found := false
+
+			for _, choice := range arg.Choices {
+				if choice == args[0] {
+					found = true
+					break
+				}
+			}
+
+			if !found {
+				allowed := strings.Join(arg.Choices[0:len(arg.Choices)-1], ", ")
```

Comment:
why not just `arg.Choices`?

Reply:
The right-hand side of the slice expression deliberately excludes the final element of slices containing more than one element so it can be appended with `, or` as a delimiter. The same logic is used upstream for [options](https://github.com/thought-machine/go-flags/blob/1dad0499af0f947f416c40f7899212e8f0c61302/option.go#L261).